src/frontend/src/pages/GanttPage/GanttChartComponents/GanttTaskBar/GanttTaskBarEditView.tsx

```diff
@@ -0,0 +1,234 @@
+      blocking: [],
+      descriptionBullets: [],
+      links: [],
+      wbsElementId: '-1',
+      dateCreated: new Date(),
+      lead: undefined,
+      manager: undefined,
+      teamTypes: [],
+      changes: [],
+      materials: [],
+      assemblies: []
+    };
+    addWorkPackage(workPackage);
+
+    createChange({
+      id,
+      type: 'create-work-package',
+      element: workPackage
+    });
+  };
+
+  // used to make sure that any changes to the start and end dates are made in multiples of 7
+  const roundToMultipleOf7 = (num: number) => {
+    return Math.round(num / 7) * 7;
+  };
+
+  const handleMouseDown = (e: MouseEvent<HTMLElement>) => {
+    setIsResizing(true);
+    setStartX(e.clientX);
+  };
+
+  const handleMouseMove = (e: MouseEvent<HTMLElement>) => {
+    if (isResizing) {
+      const currentX = e.clientX;
+      const deltaX = currentX - startX!;
+      setWidth(Math.max(100, width + deltaX));
+      setStartX(currentX);
+    }
+  };
+
+  const handleMouseUp = () => {
+    if (isResizing) {
+      setIsResizing(false);
+      // Use change in width to calculate new length
+      const newEventLengthInDays = roundToMultipleOf7(width / widthPerDay);
+      // The gantt chart tasks are inclusive (their width includes the full width of their start and end date)
+      const displayWeeks = newEventLengthInDays / 7 + 1;
+      // We need these magic pixel numbers to dynamically calculate the correct width of the task to keep it in sync with the stored end date
+      const correctWidth = displayWeeks * 38 + (displayWeeks - 1) * 10;
+      const newEndDate = addDaysToDate(task.start, newEventLengthInDays);
+      setWidth(correctWidth);
+      createChange({
+        id,
+        element: transformGanttTaskToWorkPackage(task),
+        type: 'change-end-date',
+        originalEnd: task.end,
+        newEnd: newEndDate
+      });
+    }
+  };
+
+  const onDragStart = () => {
+    setShowDropPoints(true);
+  };
+  const onDragEnd = () => {
+    setShowDropPoints(false);
+  };
+
+  const onDragOver = (e: DragEvent<HTMLDivElement>) => {
+    e.preventDefault();
+  };
+  const onDrop = (day: Date) => {
+    const days = roundToMultipleOf7(differenceInDays(day, task.start));
+    createChange({ id, element: transformGanttTaskToWorkPackage(task), type: 'shift-by-days', days });
+  };
+
+  const AddWorkPackageModal = () => {
+    return (
+      <AddGanttWorkPackageModal
+        showModal={showAddWorkPackageModal}
+        handleClose={() => setShowAddWorkPackageModal(false)}
+        addWorkPackage={handleAddWorkPackageInfo}
+      />
+    );
+  };
+
+  return (
+    <div style={ganttTaskBarContainerStyles()} onMouseUp={handleMouseUp} onMouseMove={handleMouseMove}>
+      <AddWorkPackageModal />
+      <Box
+        sx={{
+          ...ganttTaskBarBackgroundStyles(days.length),
+          position: 'absolute', // These will make it so that the bar stays on top of the drop points, i kind of like it going to a new line though
```

Comment:
oh you're applying it here, but commenting it out in the helper method? Go with whichever way you think looks better and feel free to remove the comments

Reply:
yeah it needs to be here to keep it inline. Idk which one i prefer